__tests__/app.test.js

```diff
@@ -5,32 +5,143 @@ const app = require("../app");
     return request(app)
       .get("/api/topics")
       .expect(200)
       .then((response) => {
-        expect(Array.isArray(response.body)).toBe(true);
-        expect(response.body.length).toBe(3);
-        expect(response.body[0]).toHaveProperty("slug");
-        expect(response.body[0]).toHaveProperty("description");
+        expect(Array.isArray(response.body.topics)).toBe(true);
+        expect(response.body.topics.length).toBe(3);
+
+        response.body.topics.forEach((topic) => {
+          expect(topic).toHaveProperty("slug");
+          expect(topic.slug).toEqual(expect.any(String));
+          expect(topic).toHaveProperty("description");
+          expect(topic.description).toEqual(expect.any(String));
+        });
+      });
+  });
+});
+
+describe("GET /api/", () => {
+  test("should return a 400 status code if path does not exist", () => {
+    return request(app).get("/api/space").expect(400);
+  });
+
+  test("should have a description of all endpoints", () => {
+    const endpointJson = require("../endpoints.json");
+
+    return request(app)
+      .get("/api/")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.api).toEqual(endpointJson);
+      });
+  });
+});
+
+describe("GET /api/articles/:article_id", () => {
+  test("should return a 200 status code", () => {
+    return request(app).get("/api/articles/1").expect(200);
+  });
+
+  test("should return a 404 status code for an article that doesn't exist", () => {
+    const nonExistentArticleId = 76;
+
+    return request(app)
+      .get(`/api/articles/${nonExistentArticleId}`)
+      .expect(404);
+  });
+
+  test("should return the article details for an existing article", () => {
+    const existingArticleId = 6;
+
+    return request(app)
+      .get(`/api/articles/${existingArticleId}`)
+      .expect(200)
+      .then((response) => {
+        expect(Array.isArray(response.articles)).toBe(false);
+        expect(response.body.articles[0]).toHaveProperty("author");
+        expect(response.body.articles[0]).toHaveProperty("title");
+        expect(response.body.articles[0]).toHaveProperty(
+          "article_id",
+          existingArticleId
+        );
+        expect(response.body.articles[0]).toHaveProperty("body");
+        expect(response.body.articles[0]).toHaveProperty("topic");
+        expect(response.body.articles[0]).toHaveProperty("created_at");
+        expect(response.body.articles[0]).toHaveProperty("votes");
+        expect(response.body.articles[0]).toHaveProperty("article_img_url");
       });
   });
 });
-//Checks that it returns an array
+
+test("should return the article details for an existing article", () => {
+  const anotherExistingArticleId = 1;
+
+  return request(app)
+    .get(`/api/articles/${anotherExistingArticleId}`)
+    .expect(200)
+    .then((response) => {
+      expect(Array.isArray(response.articles)).toBe(false);
+    });
+});
+
+describe("GET /api/articles/", () => {
```

Comment:
we need another 200 test to check that the articles are ordered by date descending, check out this package https://github.com/P-Copley/jest-sorted